dmithamo1/task-manager-api
# #2 Model a Task

#### What does this PR do?
This PR configures a `PostgreSQL` DB for the API and defines the fields specific to a Task
#### Description of Task to be completed
- start the project and the `tasks` app using `django-admin`
- define the schema for a Task in the DB
- refine the `PULL_REQUEST_TEMPLATE`

#### How should this be manually tested?
1. Set up and activate [virtualenv](https://virtualenv.pypa.io/)
2. Run the following commands in Terminal

    ```bash
    #!user/bin/bash
    createdb tasks_api_db
    pip install -r requirements

    # ensure you are in the folder that contains the manage.py file

    python manage.py makemigrations
    python manage.py migrate
    ```

3. Verify successful creation of the `tasks_task` table in the `tasks_api_db` with all the columns specified in `tasks/models.py` by accessing the db from a `psql` client (e.g [Postico](https://eggerapps.at/postico/) ). You can also do this in terminal as below:

    ```bash
    #!user/bin/bash
    psql tasks_api_db
    \d+ tasks_task
    ```

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
`Folder structure`

![image](https://user-images.githubusercontent.com/45358371/55901751-a752ec00-5bd2-11e9-9519-8c3aff10d809.png)


#### Checklist
